haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java

```diff
@@ -40,13 +41,16 @@ public List<Map.Entry<Character, Long>> topLetters()
         // TODO: of Character objects to their counts
         // Hint: Look at IntStream's filter, map, mapToObject, collect methods
         // Hint: Also loo at Collectors.groupingBy, Collectors.counting
-        Map<Character, Long> chars = null;
+        Map<Character, Long> chars = this.getHaikuAsChars().filter(Character::isLetter)
+                .mapToObj(i -> Character.toLowerCase((char) i)).collect(Collectors.groupingBy(c -> c, Collectors.counting()));
 
         // TODO: Sort the entries in the Map by their values in reverseOrder
         // TODO: Take the top three entries and convert them to a List
         // Hint: Look at entrySet, stream, sorted, Map.Entry.comparingByValue, Comparator.reverseOrder()
         // Hint: On Stream look at limit and toList.
-        return null;
+
+
+        return chars.entrySet().stream().sorted(Map.Entry.comparingByValue(Comparator.reverseOrder())).limit(3).collect(Collectors.toList());
```

Comment:
Please assign this result to a variable before returning to enhance code readability